lib/models/application.js

```diff
@@ -1257,28 +1217,27 @@ const getApplicationModel = function (deps, opts) {
 		 * 	if (error) throw error;
 		 * });
 		 */
-		grantSupportAccess(nameOrSlugOrId, expiryTimestamp) {
+		async grantSupportAccess(nameOrSlugOrId, expiryTimestamp) {
 			if (expiryTimestamp == null || expiryTimestamp <= Date.now()) {
 				throw new errors.BalenaInvalidParameterError(
 					'expiryTimestamp',
 					expiryTimestamp,
 				);
 			}
 
-			return getId(nameOrSlugOrId)
-				.then((applicationId) =>
-					pine.patch({
-						resource: 'application',
-						id: applicationId,
-						body: { is_accessible_by_support_until__date: expiryTimestamp },
-					}),
-				)
-				.catch((err) => {
-					if (isNotFoundResponse(err)) {
-						treatAsMissingApplication(nameOrSlugOrId, err);
-					}
-					throw err;
+			try {
+				const applicationId = await getId(nameOrSlugOrId);
+				return pine.patch({
```

Comment:
```suggestion
				return await pine.patch({
```
Otherwise the catch will not run if `pine.patch` fails, which is more of an issue if `nameOrSlugOrId` is a number, in which case `getId` doesn't check with the API.